EmployeeLib/Employee.cs

```diff
@@ -1,16 +1,17 @@
-﻿namespace EmployeeLib
+﻿using EmployeeLib.Interfaces;
+using System;
+
+namespace EmployeeLib
 {
-    public class Employee
+    public abstract class EmployeeWithManager : IEmployee, IEmployeeWithManager
     {
         public string FirstName { get; set; }
         public string LastName { get; set; }
-        public Employee Manager { get; set; } = null;
         public decimal Salary { get; set; }
+        public EmployeeWithManager Manager { get; set; }
 
-        public virtual void AssignManager(Employee manager)
+        public void AssignManager(EmployeeWithManager manager)
```

Comment:
this will result to issues when assigning a sales to be manager for a manager for example. refactor this code please